File: maxDepth.py
class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if(root != None):
            maxLev = [0]
            for i in root.children:
                maxLev.append(self.maxDepth(i))
            return 1+max(maxLev)
        return 0

File: maxDepth.java
class Solution {
    public int maxDepth(String s) {
        int nestingDepth = 0,countOfParanthesis = 0;
        
        for(int i=0;i<s.length();i++)
        {
            if(s.charAt(i) == '(')
            {
                countOfParanthesis++;
            }
            else if(s.charAt(i) == ')')
            {
                countOfParanthesis--;
            }
            
            if(countOfParanthesis > nestingDepth)
            {
                nestingDepth = countOfParanthesis;
            }
        }
        
        return nestingDepth;
    }
}
